app/assets/javascripts/reducers/columns/backlog.js

```diff
@@ -50,13 +51,24 @@ const orderByState = stories => {
   ];
 };
 
+const groupStoriesInSprints = (stories, project) => {
+  const currentSprintNumber = Iteration.getCurrentIteration(project) || 0;
+
+  return Iteration.groupBySprints(
+    stories,
+    project,
+    currentSprintNumber
+  )
+};
+
 const backlog = (state = initialState, action) => {
   switch (action.type) {
     case actionTypes.COLUMN_BACKLOG:
       const stories = [...state.stories, action.data];
 
       return {
-        stories: orderByState(stories)
+        stories: orderByState(stories),
+        sprints: groupStoriesInSprints(stories, action.project)
```

Comment:
I _think_ we need to order the stories first and then pass them to the `groupStoriesInSprints`, otherwise they'll be grouped in the wrong order, right?